securitycenter/notificationsv2/notifications_test.go

```diff
@@ -0,0 +1,265 @@
+)
+
+func orgID(t *testing.T) string {
+	orgID := os.Getenv("GCLOUD_ORGANIZATION")
+	if orgID == "" {
+		t.Skip("GCLOUD_ORGANIZATION not set")
+	}
+	return orgID
+}
+
+func projectID(t *testing.T) string {
+	projectID := os.Getenv("SCC_PUBSUB_PROJECT")
+	if projectID == "" {
+		t.Skip("SCC_PUBSUB_PROJECT not set")
+	}
+	return projectID
+}
+
+func pubsubTopic(t *testing.T) string {
+	pubsubTopic := os.Getenv("SCC_PUBSUB_TOPIC")
+	if pubsubTopic == "" {
+		t.Skip("SCC_PUBSUB_TOPIC not set")
+	}
+	return pubsubTopic
+}
+
+func pubsubSubscription(t *testing.T) string {
+	pubsubSubscription := os.Getenv("SCC_PUBSUB_SUBSCRIPTION")
+	if pubsubSubscription == "" {
+		t.Skip("SCC_PUBSUB_SUBSCRIPTION not set")
+	}
+	return pubsubSubscription
+}
+
+func addNotificationConfig(t *testing.T, notificationConfigID string) error {
+	orgID := orgID(t)
+	pubsubTopic := pubsubTopic(t)
+
+	ctx := context.Background()
+	client, err := securitycenter.NewClient(ctx)
+
+	if err != nil {
+		return fmt.Errorf("securitycenter.NewClient: %w", err)
+	}
+	defer client.Close()
+
+	req := &securitycenterpb.CreateNotificationConfigRequest{
+		Parent:   fmt.Sprintf("organizations/%s", orgID),
+		ConfigId: notificationConfigID,
+		NotificationConfig: &securitycenterpb.NotificationConfig{
+			Description: "Go sample config",
+			PubsubTopic: pubsubTopic,
+			NotifyConfig: &securitycenterpb.NotificationConfig_StreamingConfig_{
+				StreamingConfig: &securitycenterpb.NotificationConfig_StreamingConfig{
+					Filter: `state = "ACTIVE"`,
+				},
+			},
+		},
+	}
+
+	_, err0 := client.CreateNotificationConfig(ctx, req)
+	if err0 != nil {
+		return fmt.Errorf("Failed to create notification config: %w", err0)
+	}
+
+	return nil
+}
+
+func cleanupNotificationConfig(t *testing.T, notificationConfigID string) error {
+	orgID := orgID(t)
+
+	ctx := context.Background()
+	client, err := securitycenter.NewClient(ctx)
+
+	if err != nil {
+		return fmt.Errorf("securitycenter.NewClient: %w", err)
+	}
+	defer client.Close()
+
+	name := fmt.Sprintf("organizations/%s/notificationConfigs/%s", orgID, notificationConfigID)
+	req := &securitycenterpb.DeleteNotificationConfigRequest{
+		Name: name,
+	}
+
+	if err = client.DeleteNotificationConfig(ctx, req); err != nil {
+		return fmt.Errorf("Failed to retrieve notification config: %w", err)
+	}
+
+	return nil
+}
+
+func TestCreateNotificationConfig(t *testing.T) {
+	testutil.Retry(t, 5, 30*time.Second, func(r *testutil.R) {
+		buf := new(bytes.Buffer)
+		rand, err := uuid.NewUUID()
+		if err != nil {
+			r.Errorf("Issue generating id.")
+			return
+		}
+		configID := "go-test-create-config-id" + rand.String()
+
+		if err := createNotificationConfig(buf, orgID(t), pubsubTopic(t), configID); err != nil {
+			r.Errorf("createNotificationConfig failed: %v", err)
+			return
+		}
+
+		if !strings.Contains(buf.String(), "New NotificationConfig created") {
+			r.Errorf("createNotificationConfig did not create.")
+		}
+
+		cleanupNotificationConfig(t, configID)
```

Comment:
for all tests, you can `defer` the cleanup method right after the create, so that the resoruces are deleted no matter how we exit the function (e.g. if there's a nil pointer error). deferred calls happen any time we exit the method, where this is only called if we get to the end.

Reply:
Addressed